### PR TITLE
DATAUP-719: use cell model to generate job params

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Update the job manager to use the job params from the model to populate the parameters of child jobs in a batch.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-719
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative, creating and running a new bulk import job, and being amazed by the immediate population of the table with the job params!

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/jobManager.js [hunk 1]
```diff
@@ -896,22 +896,33 @@ define([
...
                         status: 'created',
                         created: 0,
                     };
                 });
+                jobInfo[batch_id] = {
+                    app_id: 'batch',
+                    app_name: 'batch',
+                    outputParams: [],
+                    userParams: null,
+                };
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobManager.js:917]: this data is never displayed anywhere, but it's put in for completeness

File: nbextensions/bulkImportCell/bulkImportCell.js [hunk 2]
```diff
@@ -604,18 +604,20 @@ define([
...
-                    tag: appSpecInfo.tag || 'release',
-                    version: appSpecInfo.git_commit_hash,
-                    params: params[fileType].filePaths.map((filePathParams) => {
-                        return Object.assign({}, filePathParams, params[fileType].params);
-                    }),
-                };
-                appInfos.push(appInfo);
-            });
+            Object.keys(inputs)
+                .sort()
```
Comment (ialarmedalien) [inline nbextensions/bulkImportCell/bulkImportCell.js:608]: This is the only difference -- added to enforce a consistent order when going through the keys.

File: kbase-extension/static/kbase/js/common/jobManager.js [hunk 3]
```diff
@@ -961,6 +972,38 @@ define([
...
+                Object.keys(inputs)
+                    .sort()
+                    .forEach((fileType) => {
+                        const appSpecInfo = appSpecs[inputs[fileType].appId].full_info;
+                        params[fileType].filePaths.forEach((filePathParams) => {
+                            const filteredParams = {
+                                app_id: appSpecInfo.id,
+                            };
+                            filteredParams.userParams = {};
+                            appSpecs[appSpecInfo.id].parameters.forEach((param) => {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobManager.js:991]: pick out the user-configurable params

File: kbase-extension/static/kbase/js/common/jobManager.js [hunk 4]
```diff
@@ -961,6 +972,38 @@ define([
...
+                                const paramId = param.id;
+                                if (filePathParams[paramId]) {
+                                    filteredParams.userParams[paramId] = filePathParams[paramId];
+                                } else if (params[fileType].params[paramId]) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobManager.js:995]: `filePathParams` are unique per job, and the `params[fileType].params` are shared between all of an app's jobs

File: kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStatusTable.js [hunk 5]
```diff
@@ -121,26 +121,19 @@ define([
...
-                    });
+                Object.values(appData.specs[jobInfo.app_id].parameters).forEach((param) => {
+                    const paramId = param.id;
+                    // params generated by the job manager
+                    if (jobInfo.userParams && jobInfo.userParams[paramId]) {
+                        const value = jobInfo.userParams[paramId];
+                        const label = param.ui_name;
+                        params[paramId] = { value, label };
+                        // params from a job info call
+                    } else if (jobInfo.job_params && jobInfo.job_params[0][paramId]) {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStatusTable.js:132]: this is a nuisance -- we don't want to have two different formats for the job info, but there are already existing narratives with info in this format. I will unify the format in another PR.

**Review by briehl — APPROVED** [review]
👍

**Pull request merged**